src/webgpu/util/floating_point.ts

```diff
@@ -708,6 +709,41 @@ export abstract class FPTraits {
     return new FPInterval(this.kind, n, n);
   }
 
+  /**
+   * WGSL specifies unbounded precision:
+   *   https://www.w3.org/TR/WGSL/#floating-point-accuracy
+   * In most computations doubles (number in js) are equivalent
+   * to correctly rounded intervals. However in some cases
+   * (addition/subtraction) doubles do not provide enough numeric precision.
+   *
+   * These cases are whenever a number, x (e.g. 1.0) is added (or subtracted)
+   * from a much smaller number, y (e.g. smallest positive normal), such that
+   * the difference between them is smaller then ULP(x). When working in JS
+   * numbers the result will, incorrectly, simply be x, since it will get
+   * rounded. We must detect these cases where double precision does not
+   * represent infinitely accurate computations accurately then we must
+   * manually create the interval.
+   */
+  /** @returns an interval containing the correctly rounded val with respect to unbounded precision */
+  public correctlyRoundedIntervalWithUnboundedPrecision(
+    val: number,
```

Comment:
I recommend documenting the parameters:

@param sum the result of adding large_val + small_val, using Number
@param large_val: the summand with larger magnitude
@param small_val: the summand with smaller magnitude